app/main.py

```diff
@@ -1,6 +1,18 @@
 from typing import Callable
+from functools import wraps
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    cache = {}
+
+    @wraps(func)
+    def wrapper(*args, **kwargs) -> Callable:
+        key = (args, frozenset(kwargs.items()))
```

Comment:
Use name of function as part of `key` tuple since we can face with case when different functions have same arguments